### PR TITLE
Manage Temp Files - New Doc

### LINKED ISSUE #1172: Create doc on how to handle temp files
Issue:
Customers need an alternative to a /tmp directory.

Solution:
Create a new doc that instructs customers the best alternatives to using /tmp.

Possibilities:
* Use Valhalla. If you do, how do you do it so it works, what use cases are acceptable and which ones won't ever work?
* Other workaround/solution for a /tmp directory location
* Specify a single appserver until that job is completed 
* (from power users thread) There's a way to use plupload successfully with Pantheon. Basically, in filefield_sources_plupload.module, you just need to change the default upload directory from tmp to private://tmp (you might also need to write something to run through files/private/tmp every few hours and delete old files to keep it from piling up). Here's the line that did it for me:
$temp_destination = file_stream_wrapper_uri_normalize('private://tmp/' . $filename);
That moves the temporary upload destination from the individual server mount tmp directory to the shared mount tmp files/private/tmp directory, which should preserve the files between requests.

TODO

- [x] Research how to manage /tmp files on distributed infrastructure
- [ ] Check with CSE for workarounds/suggestions
- [ ] Confirm suggestions w/ Josh

- Requested by @sukottokun 
See Power Users thread Multi-Upload files
and this issue  https://github.com/pantheon-systems/documentation/pull/584

### PR DESCRIPTION
**Pull request by nataliejeremy**
This closes #1172 

`@cse` please help with the single appserver section and add any other workarounds you can think of. Thanks!

### CODE REVIEW

File: source/docs/articles/sites/files/temp-files.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+description: Learn how to work with temporary files in distributed environments.
+category:
+- getting-started
+keywords: tmp, temp files, tmp files, temporary files, multiple application containers, distributed environments
+---
+Live sites on Professional plans and above have multiple [application containers](/docs/articles/sites/all-about-application-containers).  Pantheon's distributed system means that requests are spread between all of the available application servers, which is part of how we help the site scale. However, the `/tmp` directory on one instance is not able to access the `/tmp` resources on another application server. There are several solutions available that allow you to upload files and create temporary directories on Pantheon.
+
+## Plupload
+
+ [Plupload](http://www.plupload.com/) is a multi-runtime file uploader and has been used successfully with the Pantheon platform. To do so, change the default upload directory from `tmp` to `private://tmp` in  `filefield_sources_plupload.module`. You may also need to add this line to run through `files/private/tmp` every few hours and delete old files to keep it from piling up:
```
Comment (ari-gold) [inline source/docs/articles/sites/files/temp-files.md]: Is there a patch we can link to on drupal.org?

Comment (nataliejeremy) [inline source/docs/articles/sites/files/temp-files.md]: @ari-gold Think I should link to this one? https://www.drupal.org/project/plupload

**Comment by bmackinney** [comment]
`@cse`

File: source/docs/articles/sites/files/temp-files.md (hunk 1 above)
Comment (bmackinney) [inline source/docs/articles/sites/files/temp-files.md]: @ari-gold The issue/solution is from https://www.drupal.org/node/1184850. I don't thikn there's a patch. Do you think we should adapt Jen's Acquia snippet (final comment) to Pantheon?

File: source/docs/articles/sites/files/temp-files.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Plupload
+
+ [Plupload](http://www.plupload.com/) is a multi-runtime file uploader and has been used successfully with the Pantheon platform. To do so, change the default upload directory from `tmp` to `private://tmp` in  `filefield_sources_plupload.module`. You may also need to add this line to run through `files/private/tmp` every few hours and delete old files to keep it from piling up:   
+ `$temp_destination = file_stream_wrapper_uri_normalize('private://tmp/' . $filename);`  
+ This will move the temporary upload destination from the individual server mount `tmp` directory to the shared `mount tmp files/private/tmp directory`, which should preserve the files between requests.
+
+## Specify a Single Appserver
```
Comment (bmackinney) [inline source/docs/articles/sites/files/temp-files.md]: @timani please meet with @nataliejeremy to complete this section.

**Comment by nataliejeremy** [comment]
@aleszoulek @zepner Can you review and help complete the section: Specify a Single Appserver

File: source/docs/articles/sites/files/temp-files.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+ This method allows you to set a specific appserver until the job is complete.  
+ Steps:  
+ 1.
```
Comment (rachelwhitton) [inline source/docs/articles/sites/files/temp-files.md]: WordPress: 
```
// Replace 'some_tmp_settings' with the plugin or theme option 
update_option('some_tmp_settings', $_SERVER['HOME'] . '/tmp');
``` 

 Drupal: 
```
// Replace 'some_tmp_settings' with the module or theme setting
$conf['some_tmp_settings'] = $_SERVER['HOME'] . '/tmp';
```

**Comment by rachelwhitton** [comment]
+1

**Comment by zepner** [comment]
+1

**Pull request merged**